cmd/infracost/run.go

```diff
@@ -375,37 +384,52 @@ func (r *parallelRunner) run() ([]projectResult, error) {
 			}()
 
 			for job := range jobs {
-				var configProjects *projectOutput
-				ctx := job.ctx
-				if job.err != nil {
-					configProjects = newErroredProject(job.provider, job.ctx, job.err)
-				} else {
-					configProjects, err = r.runProvider(job)
-					ctx = job.provider.Context()
-					if err != nil {
-						configProjects = newErroredProject(job.provider, ctx, err)
+				func() {
```

Comment:
Could you please educate me why the func is needed here? Is it because you need to defer the timer stopping?